### ISSUE TITLE
when switching browser tab or minimizing,it will pause.

### ISSUE THREAD

**Issue by king2015** [rodri042/nestation#1]
I just learned react, Where can I change it,can you help me？ @rodri042

**Comment by rodri042** [comment]
Hey, I'm not sure if I understand your problem. Currently, minimizing the window or selecting another tab will make the emulator stop, since `requestAnimationFrame` won't be called anymore.

It's not possible to keep running the game because the browser prevents that kind of behaviors to optimize computer's resources, and also there's no reason to do so.

The *master* node (the one who uploads the rom) will always handle the timing and send the state to the *slave* node. If what you want is to pause the master node when the slave node is minimized, you want to block the responses of the [SlaveSyncer](https://github.com/rodri042/nestation/blob/master/src/networking/SlaveSyncer.js#L112) when [the window is minimized](https://stackoverflow.com/questions/7389328/detect-if-browser-tab-has-focus), but there's no React code involved there.

**Comment by king2015** [comment]
> Hey, I'm not sure if I understand your problem. Currently, minimizing the window or selecting another tab will make the emulator stop, since `requestAnimationFrame` won't be called anymore.
> 
> It's not possible to keep running the game because the browser prevents that kind of behaviors to optimize computer's resources, and also there's no reason to do so.
> 
> The _master_ node (the one who uploads the rom) will always handle the timing and send the state to the _slave_ node. If what you want is to pause the master node when the slave node is minimized, you want to block the responses of the [SlaveSyncer](https://github.com/rodri042/nestation/blob/master/src/networking/SlaveSyncer.js#L112) when [the window is minimized](https://stackoverflow.com/questions/7389328/detect-if-browser-tab-has-focus), but there's no React code involved there.

but why the jsnes website can keep running the game
I tried it on https://jsnes.org

**Comment by rodri042** [comment]
jsnes drives the emulator by audio when the tab is minimized. Though it's a
clever hack, it's not so accurate and causes several bugs (e.g. it doesn't
work with 120hz monitors) so I've removed that part.

El 29 jul. 2019 3:37 a. m., "king2015" <notifications@github.com> escribió:

Hey, I'm not sure if I understand your problem. Currently, minimizing the
window or selecting another tab will make the emulator stop, since
requestAnimationFrame won't be called anymore.

It's not possible to keep running the game because the browser prevents
that kind of behaviors to optimize computer's resources, and also there's
no reason to do so.

The *master* node (the one who uploads the rom) will always handle the
timing and send the state to the *slave* node. If what you want is to pause
the master node when the slave node is minimized, you want to block the
responses of the SlaveSyncer
<https://github.com/rodri042/nestation/blob/master/src/networking/SlaveSyncer.js#L112>
when the window is minimized
<https://stackoverflow.com/questions/7389328/detect-if-browser-tab-has-focus>,
but there's no React code involved there.

but why the jsnes website can keep running the game
I tried it on https://jsnes.org

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/rodri042/nestation/issues/1?email_source=notifications&email_token=AAMOMCBPVJS2BRHWVWWVEPDQB2F4DA5CNFSM4IHMHA6KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD27XQOY#issuecomment-515864635>,
or mute the thread
